extensions/wikia/Helios/User.class.php

```diff
@@ -198,6 +198,78 @@ public static function onUserCheckPassword( $id, $username, $hash, $password, &$
 		return true;
 	}
 
+    /**
+     * Called in ExternalUser_Wikia registers a user.
+     *
+     * @param string $username string of the user name
+     * @param string $password string of the plaintext password the user entered
+     * @param string $email string of the user email
+     *
+     * @return boolean true on success, false otherwise
+     */
+    public static function register( $username, $password, $email )
+    {
+        $logger = \Wikia\Logger\WikiaLogger::instance();
+        $logger->info( 'HELIOS_REGISTRATION', [ 'method' => __METHOD__ ] );
+
+        global $wgHeliosBaseUri, $wgHeliosClientId, $wgHeliosClientSecret;
+        $helios = new Client( $wgHeliosBaseUri, $wgHeliosClientId, $wgHeliosClientSecret );
+
+        try {
+            $registration = $helios->register( $username, $password, $email );
+            $result = !empty( $registration->success );
+
+            if ( !empty( $registration->error ) ) {
+                $logger->error(
+                    'HELIOS_REGISTRATION',
+                    [ 'method' => __METHOD__ ]
+                );
+            }
+        }
+
+        catch ( ClientException $e ) {
+            $logger->error(
+                'HELIOS_REGISTRATION',
+                [ 'exception' => $e, 'method' => __METHOD__ ]
+            );
+            $result = false;
+        }
+
+        return $result;
+    }
+
+    public static function onRegister( &$result, &$userId, $User, $password, $email ) {
+
+        $heliosUserId = null;
+        $heliosResult = self::register( $User->mName, $password, $email, $User->mBirthDate );
+        $logger = \Wikia\Logger\WikiaLogger::instance();
+
+        if ( $heliosResult ) {
+
+            global $wgExternalSharedDB;
+            $dbw = \wfGetDB( DB_SLAVE, [], $wgExternalSharedDB );
```

Comment:
in order to remove the possible race condition of slave lag I'd use DB_MASTER here